src/game/movement.rs

```diff
@@ -1,5 +1,12 @@
-//! Handle player input and translate it into movement.
-//! Note that the approach used here is simple for demonstration purposes.
+//! Handle player input and translate it into movement through a character controller.
+/// A character controller is the collection of systems that govern the movement of characters.
+//!
+/// In our case, the character controller has the following logic:
+/// - Set the movement intent based on directional input.
+/// - Apply movement based on the intent.
+/// - Wrap the character within the window.
+///
+//! Note that the implementation used here is simple for demonstration purposes.
```

Comment:
```suggestion
//! Note that the implementation used here is limited for demonstration purposes.
```
To avoid the "simple => you should understand it easily" misunderstanding.